app/src/main/java/com/parking_system_kotlin/activities/ReservationActivity.kt

```diff
@@ -30,11 +32,21 @@ class ReservationActivity : AppCompatActivity(), ListenerDateTime {
             editTextReservationActivityEntry.setOnClickListener { presenter.showDatePicker(this@ReservationActivity, true) }
             editTextReservationActivityExit.setOnClickListener { presenter.showDatePicker(this@ReservationActivity, false) }
             buttonReservationActivitySave.setOnClickListener {
-                presenter.saveReservation()
+                makeReservation()
             }
         }
     }
 
+    private fun makeReservation() {
+        presenter.saveReservation(
+            Reservation(
```

Comment:
This should be in the presenter. The activity should call the presenter and from there should ask the view al the information to create the Reservation

Reply:
done